staging/src/k8s.io/kubectl/pkg/util/podutils/podutils.go

```diff
@@ -142,18 +146,26 @@ func (s ActivePods) Less(i, j int) bool {
 	if IsPodReady(s[i]) != IsPodReady(s[j]) {
 		return !IsPodReady(s[i])
 	}
+	// 4. Deleting < Not deleting
+	if isPodDeleting(s[i]) != isPodDeleting(s[j]) {
+		return isPodDeleting(s[i])
+	}
+	// 5. Older deletion timestamp < newer deletion timestamp
+	if isPodDeleting(s[i]) && isPodDeleting(s[j]) && !s[i].ObjectMeta.DeletionTimestamp.Equal(s[j].ObjectMeta.DeletionTimestamp) {
```

Comment:
I think we can remove `!s[i].ObjectMeta.DeletionTimestamp.Equal(s[j].ObjectMeta.DeletionTimestamp)` condition for the simplicity because it is very rare that `s[i].ObjectMeta.DeletionTimestamp` and `s[j].ObjectMeta.DeletionTimestamp`  are equal and even this happens, we can continue to other conditions enumerated below.

Reply:
Thanks for that comment.

The reason I original added that condition was to follow the pattern set by the comparison of ready time below (step 6).

At first, I agreed with your comment and updated this PR, but then I updated it back because I'm concerned that if pod deletion timestamp is not nil and also equal, it won't fall through to the other conditions then, and it will arbitrarily choose one of the deleting pods without any consideration for the other factors.

Removing that condition could make the active pod comparisons "unstable", in a stable-sort sort of way... By that, I mean calling it multiple times with the same elements in a different starting order could yield a different result, when ideally it should be the same result regardless of the input order.

I agree that it seems unlikely for two pods to have **exactly** the same deletion timestamp, but it could happen with an automated process that bulk-deletes pods for example.

Anyway, I think it seems safer to leave this condition in.  I think it is worth the reduced simplicity to be explicit about how this case is handled.

My two pushes (change and change back) removed your lgtm, so if you still think this is OK, another lgtm would be appreciated! :sweat_smile:

Reply:
It makes sense and I agree with that(especially comparing with the step 6 which applies similar rule);
/lgtm